tests/Elasticsearch/Tests/ConnectionPool/Selectors/RoundRobinSelectorTest.php

```diff
@@ -52,33 +66,39 @@ public function testTenConnections()
      *
      * @return void
      */
-    public function testAddTenConnectionsestFiveTRemoveThree()
+    public function testAddTenConnectionsTestFiveRemoveThreeTestTen()
     {
         $roundRobin = new Elasticsearch\ConnectionPool\Selectors\RoundRobinSelector();
 
-        $mockConnections = array();
-        foreach (range(0, 10) as $index) {
-            $mockConnections[$index] = $this->getMockBuilder('\Elasticsearch\Connections\CurlMultiConnection')
+        $mockConnections = [];
+        foreach (range(0, 9) as $index) {
+            $mockConnections[$index] = $this->getMockBuilder(ConnectionInterface::class)
                 ->disableOriginalConstructor()
                 ->getMock();
         }
 
-        foreach (range(0, 4) as $index) {
-            $retConnection = $roundRobin->select($mockConnections);
-
-            $nextIndex = ($index % (count($mockConnections)-1)) + 1;
-            $this->assertEquals($mockConnections[$nextIndex], $retConnection);
-        }
+        // select five
```

Comment:
Same as above - I replaced quite complicated (and possibly broken) test logic with a straightforward code.